streams/src/main/java/org/apache/kafka/streams/processor/internals/StreamThread.java

```diff
@@ -1147,6 +1162,10 @@ public String toString(final String indent) {
         return indent + "\tStreamsThread threadId: " + getName() + "\n" + taskManager.toString(indent);
     }
 
+    public String getGroupInstanceID() {
```

Comment:
Why not an `Optional`?

Reply:
Either way we need to deal with it. I thought it would be easier to just do it once. But It probably better practice to handle it later. I will change it to Optional